File: using-nodejs/secret_conn.js
// Use this code snippet in your app.
// If you need more information about configurations or implementing the sample code, visit the AWS docs:
// https://aws.amazon.com/developers/getting-started/nodejs/

// Load the AWS SDK
var AWS = require('aws-sdk'),
    region = "", // your secret region
    secretName = "", // your secret name
    secret,
    decodedBinarySecret;

const dotenv = require('dotenv');
dotenv.config();
 
// Create a Secrets Manager client
var client = new AWS.SecretsManager({
    region: region,
    secretAccessKey: process.env.SECRET_ACCESS_KEY, // add your access keys in env file
    accessKeyId: process.env.ACCESS_KEY_ID // add your access keys in env file
});

// In this sample we only handle the specific exceptions for the 'GetSecretValue' API.
// See https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html
// We rethrow the exception by default.

client.getSecretValue({SecretId: secretName}, function(err, data) {
    if (err) {
        console.log(err)
        if (err.code === 'DecryptionFailureException')
            // Secrets Manager can't decrypt the protected secret text using the provided KMS key.
            // Deal with the exception here, and/or rethrow at your discretion.
            throw err;
        else if (err.code === 'InternalServiceErrorException')
            // An error occurred on the server side.
            // Deal with the exception here, and/or rethrow at your discretion.
            throw err;
        else if (err.code === 'InvalidParameterException')
            // You provided an invalid value for a parameter.
            // Deal with the exception here, and/or rethrow at your discretion.
            throw err;
        else if (err.code === 'InvalidRequestException')
            // You provided a parameter value that is not valid for the current state of the resource.
            // Deal with the exception here, and/or rethrow at your discretion.
            throw err;
        else if (err.code === 'ResourceNotFoundException')
            // We can't find the resource that you asked for.
            // Deal with the exception here, and/or rethrow at your discretion.
            throw err;
    }
    else {
        // Decrypts secret using the associated KMS CMK.
        // Depending on whether the secret is a string or binary, one of these fields will be populated.
        if ('SecretString' in data) {
            console.log(data)
            secret = data.SecretString;
        } else {
            let buff = new Buffer(data.SecretBinary, 'base64');
            console.log(decodedBinarySecret)
            decodedBinarySecret = buff.toString('ascii');
        }
    }

    // Your code goes here. 
    
    //Parsing secret JSON object
    const secretJSON = JSON.parse(secret);
    const { Pool } = require('pg')
    //Pass connection info to db
    //As we are using Postgres DB, 'pg' module is used, in case of MYSQL, you can connect in similar way by using 'mysql' module
    const pool = new Pool({
        user: secretJSON.username,
        host: secretJSON.host,
        database: secretJSON.dbname,
        password: secretJSON.password,
        port: 5432,
    })
    //Query the table, Replace table name with your table name
    pool.query('SELECT * from user_login limit 1', (err, res) => {
        console.log(res.rows) 
        pool.end() 
    })
});

File: using-python/secret_conn.py
# Use this code snippet in your app.
import boto3
from botocore.exceptions import ClientError
import json
import base64
import psycopg2

def get_secret():
    secret_name = "" #your secret name
    region_name = "" #your secret region

    # Create a Secrets Manager client
    session = boto3.session.Session()
    client = session.client(
        service_name='secretsmanager',
        region_name=region_name,
        aws_access_key_id = '', #add your access keys
        aws_secret_access_key = '' #add your access keys
    )

    try:
        get_secret_value_response = client.get_secret_value(
                SecretId=secret_name
            )
        
        print("Response",get_secret_value_response)
    except ClientError as e:
        if e.response['Error']['Code'] == 'ResourceNotFoundException':
            print("The requested secret " + secret_name + " was not found")
        elif e.response['Error']['Code'] == 'InvalidRequestException':
            print("The request was invalid due to:", e)
        elif e.response['Error']['Code'] == 'InvalidParameterException':
            print("The request had invalid params:", e)
    else:
        # Decrypted secret using the associated KMS CMK
        # Depending on whether the secret was a string or binary, one of these fields will be populated
        if 'SecretString' in get_secret_value_response:
            secret = get_secret_value_response['SecretString']
            print("Secret JSON : ",secret)
            # Get the user name, password, and database connection information from a config file.
            database = json.loads(secret)['dbname']
            user_name = json.loads(secret)['username']
            password = json.loads(secret)['password']
            host = json.loads(secret)['host']

            # Use the user name, password, and database connection information to connect to the database
	        #As we are using Postgres DB, 'psycopg2' module is used, in case of MYSQL, you can connect in similar way by using 'mysql.connector'
            db = psycopg2.connect(host=host,
                                    database=database,
                                    user=user_name,
                                    password=password)
	    
            #Query the table, Replace table name with your table name        
	        cursor = db.cursor()
            cursor.execute("Select * from user_login limit 1")
            print("Total Rows : ",cursor.rowcount)
            for row in cursor.fetchall():
               print("row",row)
            db.close()
        else:
            # binary_secret_data = get_secret_value_response['SecretBinary']
            secret = base64.b64decode(get_secret_value_response['SecretBinary'])

    return json.loads(secret)  # returns the secret as dictionary


